tests/test_segment.py

```diff
@@ -22,6 +23,108 @@
 __EXAMPLE_FILE = os.path.join('tests', 'data', 'test1_22050.wav')
 
 
+def test_cross_similarity():
+
+    def __test(n, k, metric):
+        srand()
+        # Make a data matrix
+        data_from = np.random.randn(3, n)
+        data_to = np.random.randn(3, n + 2)
+
+        D = librosa.segment.cross_similarity(data_from, data_to, k=k, metric=metric)
+
+        if k is not None:
+            real_k = min(k, n)
+            assert not np.any(D.sum(axis=1) != real_k)
+
+    for n in [20, 250]:
+        for k in [None, n // 4]:
+            for metric in ['l2', 'cosine']:
+                tester = __test
+                yield tester, n, k, metric
+
+
+def test_cross_similarity_sparse():
+
+    srand()
+    data_from = np.random.randn(3, 50)
+    data_to = np.random.randn(3, 100)
+    D_sparse = librosa.segment.cross_similarity(data_from, data_to, sparse=True)
+    D_dense = librosa.segment.cross_similarity(data_from, data_to, sparse=False)
+
+    assert scipy.sparse.isspmatrix(D_sparse)
+    assert np.allclose(D_sparse.todense(), D_dense)
+
+
+def test_cross_similarity_distance():
+
+    srand()
+    data_from = np.random.randn(3, 50)
+    data_to = np.random.randn(3, 70)
+    distance = cdist(data_from.T , data_to.T, metric='sqeuclidean')
+    rec = librosa.segment.cross_similarity(data_from, data_to, mode='distance',
+                                            metric='sqeuclidean',
+                                            sparse=True)
+
+    i, j, vals = scipy.sparse.find(rec)
+    assert np.allclose(vals, distance[i, j])
+
+
+def test_cross_similarity_affinity():
+
+    def __test(metric, bandwidth):
+        srand()
+        data_from = np.ones((3, 50))
+        data_to = np.ones((3, 70))
+        distance = cdist(data_to.T, data_from.T, metric=metric)
+        rec = librosa.segment.cross_similarity(data_from, data_to, mode='affinity',
+                                                metric=metric,
+                                                sparse=False,
+                                                bandwidth=bandwidth)
+        print(rec.shape)
+        # i, j, vals = scipy.sparse.find(rec)
+        # print(vals.shape)
+        logvals = np.log(rec + 1.0)
+        print(logvals.shape)
+
+        # After log-scaling, affinity will match distance up to a constant factor
+        distance_nonzero = np.ones(distance.shape)
+        distance_nonzero[distance > 0] = distance[distance > 0]
+        ratio = -logvals / (distance + 1.0)
+        print(ratio.shape)
+        if bandwidth is None:
+            assert np.allclose(ratio, ratio[0])
+        else:
+            print(ratio)
+            assert np.allclose(ratio, bandwidth)
+
+    for metric in ['sqeuclidean', 'cityblock']:
+        for bandwidth in [None, 1]:
+            yield __test, metric, bandwidth
```

Comment:
can you rewrite this using fixtures on `metric` and `bandwidth` instead of using yield tests?